server/mailet/mailets/src/test/java/org/apache/james/transport/matchers/AtMostPriorityTest.java

```diff
@@ -21,48 +21,33 @@
 import java.util.Collection;
 import javax.mail.MessagingException;
 import org.apache.james.core.MailAddress;
-import org.apache.james.queue.api.MailPrioritySupport;
 import org.apache.mailet.base.test.FakeMail;
-import org.apache.mailet.base.test.FakeMatcherConfig;
 import static org.assertj.core.api.Assertions.assertThat;
 import org.junit.Before;
 import org.junit.Test;
 
-public class AtMostPriorityTest {
-    private AtMostPriority matcher;
-    private FakeMail fakeMail;
-    private MailAddress testRecipient;
+public class AtMostPriorityTest extends PriorityTest {
```

Comment:
We tend to avoid complex class hierarchy and inheritance in test.

To understand the behaviour of this test class, I should understand 2 classes that are strongly linked together which is a significant cognitive overhead.

IMO the tests were easier to understand before.